src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -878,7 +890,10 @@ DagState uniquifyState(ImmutableArray<StateForCase> cases, ImmutableDictionary<B
                     {
                         // There is a when clause to evaluate.
                         // In case the when clause fails, we prepare for the remaining cases.
-                        var stateWhenFails = state.Cases.RemoveAt(0);
+                        var stateWhenFails = ArrayBuilder<StateForCase>.GetInstance(state.Cases.Count - 1);
```

Comment:
The intent in the old code was clearer. Can we keep that level of abstraction (maybe with a helper method on `FrozenArrayBuilder`)?

Reply:
Done.